lib/gcloud/datastore/proto.rb

```diff
@@ -82,6 +86,9 @@ def self.to_proto_value value
           v.string_value = value
         elsif Array === value
           v.list_value = value.map { |item| to_proto_value item }
+        elsif value.respond_to?(:read) && value.respond_to?(:rewind)
+          value.rewind
+          v.blob_value = Base64.encode64(value.read)
```

Comment:
Decoding base64 will skip the line feeds in the encoded string, but it will save a few bytes over the wire. I'll make the change.